dali/kernels/imgproc/convolution/separable_convolution_gpu_test.cu

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <cmath>
+#include <complex>
+#include <tuple>
+#include <vector>
+
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/baseline_convolution.h"
+#include "dali/kernels/imgproc/convolution/separable_convolution_gpu.h"
+#include "dali/kernels/imgproc/convolution/separable_convolution_cpu.h"
+#include "dali/kernels/scratch.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+
+/**
+ * @brief Simple sanity test for SeparableConvolutionGPU
+ */
+template <int Axes, bool Channels, bool Frames>
+class SepearableConvolutionGpuTestImpl {
+  constexpr static int kAxes = Axes;
+  constexpr static int kChannels = Channels;
+  constexpr static int kFrames = Frames;
+  constexpr static int kNdim = kFrames + kAxes + kChannels;
+
+  // Predefined window shape per axis
+  int win_sizes[3] = {3, 5, 7};
+
+  std::array<TestTensorList<float, 1>, kAxes> kernel_window_;
+  std::array<TensorListShape<1>, kAxes> window_dims_;
+  TestTensorList<float, kNdim> input_;
+  TestTensorList<float, kNdim> output_;
+  TestTensorList<float, kNdim> baseline_output_;
+  TensorListShape<kNdim> data_shape_;
+  int num_samples_;
+
+  void SetDataShape() {
+    TensorShape<> target_shape = {64, 64, 64};
+    target_shape = target_shape.last(kAxes);
+    if (kChannels)
+      target_shape = shape_cat(target_shape, 3);
+    if (kFrames)
+      target_shape = shape_cat(20, target_shape);
+    data_shape_ = uniform_list_shape<kNdim>(1, target_shape.to_static<kNdim>());
+  }
+
+  void ReshapeData() {
+    SetDataShape();
+    int num_samples_ = data_shape_.size();
+
+    for (int i = 0; i < kAxes; i++) {
+      window_dims_[i] = uniform_list_shape<1>(num_samples_, {win_sizes[i]});
+      kernel_window_[i].reshape(window_dims_[i]);
+    }
+
+    input_.reshape(data_shape_);
+    output_.reshape(data_shape_);
+    baseline_output_.reshape(data_shape_);
+  }
+
+  void FillData() {
+    ConstantFill(input_.cpu(), 1);
```

Comment:
This is not a particularly strong test....

Reply:
It's not, it's mostly sanity test as the actual testing is done in the ConvolutionGpu and this is simple wrapper. The Operator brings the full load of tests in python as well.

Reply:
Eh, added yet another similar test.